### PR TITLE
Users/ajya/create physical path for va

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/Common/webdeployment-common/kuduutility.ts [hunk 1]
```diff
@@ -133,4 +133,58 @@ export async  function containsParamFile(webAppPackage: string ) {
...
+
+    httpObj.send('GET', kuduPhysicalpathUrl, null, headers, async (error, response, body) => {
+        if (error) {
+            defer.reject(error);
+        }
+        else if (response.statusCode === 200 || response.statusCode === 201 || response.statusCode === 204) {
+            tl.debug("Physical path '" + physicalPath + "' already exists ");
+            defer.resolve(tl.loc('Physicalpathalreadyexists'));
+        }
+        else if(response.statusCode === 404) {
```
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/kuduutility.ts]: can u add a debug log as, physical path doesn't exist, creating the path ?

File: Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -87,5 +87,11 @@
   "loc.messages.UnknownFileEncodeError": "Unable to detect encoding of the file %s (typeCode: %s). Supported encodings are UTF-8 and UTF-16 LE.",
   "loc.messages.ShortFileBufferError": "File buffer is too short to detect encoding type : %s",
   "loc.messages.FailedToUpdateAzureRMWebAppConfigDetails": "Failed to update Web App config details. Error: %s",
-  "loc.messages.SuccessfullyUpdatedAzureRMWebAppConfigDetails": "Successfully updated Web App config details"
+  "loc.messages.SuccessfullyUpdatedAzureRMWebAppConfigDetails": "Successfully updated Web App config details",
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson]: App service configuration

File: Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts [hunk 3]
```diff
@@ -112,6 +112,18 @@ async function run() {
             tl.setVariable(webAppUri, publishingProfile.destinationAppUrl);
         }
 
+        var azureWebAppDetails = null;
+        if(virtualApplication) {
+            azureWebAppDetails = await azureRESTUtility.getAzureRMWebAppConfigDetails(endPoint, webAppName, resourceGroupName, deployToSlotFlag, slotName);
+            var virtualApplicationMappings = azureWebAppDetails.properties.virtualApplications;
+            var pathMappings = kuduUtility.getVirtualAndPhysicalPaths(virtualApplication, virtualApplicationMappings);
+            if("/site/wwwroot" != pathMappings[1]) {
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts]: compare with value when path is not created

File: Tasks/Common/webdeployment-common/kuduutility.ts [hunk 4]
```diff
@@ -133,4 +133,58 @@ export async  function containsParamFile(webAppPackage: string ) {
...
+        'If-Match': "*"
+    };
+    tl.debug("Requested URL for kudu physical path : " + kuduPhysicalpathUrl);
+    httpObj.send('PUT', kuduPhysicalpathUrl, null, headers, (error, response, body) => {
+        if (error) {
+            defer.reject(error);
+        }
+        else if (response.statusCode === 200 || response.statusCode === 201 || response.statusCode === 204) {
+            tl.debug("Kudu physical path : '" + physicalPath + "' created successfully ");
+            defer.resolve(tl.loc('KuduPhysicalpathCreatedSuccessfully', physicalPath));
```
Comment (mvvsubbu) [inline Tasks/Common/webdeployment-common/kuduutility.ts]: Do we need to show this message to user? I don't think it is required and hence localization not required

**Pull request merged**